server/bot/commands/utilities.ts

```diff
@@ -518,3 +518,66 @@ export const handleEditLastSay: Command = {
     } catch (e) { return 'Nothing to edit.' }
   }
 }
+
+export const handleSuppress: Command = {
+  name: 'suppress',
+  opts: {
+    requirements: {
+      permissions: { manageMessages: true }
+    },
+    description: 'Suppress or unsuppress embeds in a message.',
+    fullDescription: 'Suppress embeds in a message. If the message already has a suppressed embed, it will get unsuppressed. Requires \'Manage Messages\' permission',
+    usage: '/suppress (channel) <message ID or link>',
+    example: '/suppress #general 123456789012345678'
+  },
+  generator: async (message, args, { client }) => {
+    if (args.length === 1) {
+      let msg
+      let channelToEdit
+      const regex = /https?:\/\/((canary|ptb|www).)?discord(app)?.com\/channels\/\d{17,18}\/\d{17,18}\/\d{17,18}/
+      if (regex.test(args[0])) {
+        if (!((message.member.guild.channels.get(args[0].split('/')[5]) as TextChannel).messages.get(args[0].split('/')[6]))) {
+          msg = await client.getMessage(args[0].split('/')[5], args[0].split('/')[6])
+          channelToEdit = args[0].split('/')[5]
+        } else {
+          msg = (message.member.guild.channels.get(args[0].split('/')[5]) as TextChannel).messages.get(args[0].split('/')[6])
+          channelToEdit = args[0].split('/')[5]
+        }
+      } else {
+        if (!(message.channel.messages.get(args[0]))) {
+          msg = await message.channel.getMessage(args[0])
+          channelToEdit = message.channel.id
+        } else {
+          msg = message.channel.messages.get(args[0])
+          channelToEdit = message.channel.id
+        }
+      }
```

Comment:
There is so much redundant code here, managed to clean up a lot of it. Moved channelToEdit outside the braces and replaced the `if (obj1) a = obj1 else a = obj2` with `a = obj1 || obj2`, cleaner and faster. Also made use of variables that don't have to be avoided and help clean up code.
```suggestion
      if (regex.test(args[0])) {
        const split = args[0].split('/')
        const channel = message.member.guild.channels.get(split[5]) as GuildTextableChannel
        msg = channel.messages.get(split[6]) || await client.getMessage(split[5], split[6])
        channelToEdit = split[5]
      } else {
        msg = message.channel.messages.get(args[0]) || await message.channel.getMessage(args[0])
        channelToEdit = message.channel.id
      }
```